### ISSUE TITLE
exported: handle blank line case differently

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1234]
```go
package sandbox

// Foo is OK
func Foo() {}

// Bar is not OK because of the extra blank line

func Bar() {}
```

![Image](https://github.com/user-attachments/assets/ae866019-3218-41c1-9a5d-52f2f980b9d0)

Here everything is fine with Bar except the blank line. It could have been added randomly.

revive reports the right things yet, but it could be improved.

The godoc is empty in such a case. It could be a real problem in terms of documentation.

But, because golangci-lint exclude [`EXC0012`](https://golangci-lint.run/usage/false-positives/#exc0012) by default.
This one is not reported.

We should consider trying to detect it, and report another message that won't be caught by `EXC0012`

**Comment by denisvmedia** [comment]
In our codebase (closed source) we do have such cases, where theere is an intentional comment preceding an exported symbol, which is not a godoc comment for the symbol. What do you propose in such a case?

**Comment by ccoVeille** [comment]
It's something I had already thought about.

I also have the same in my code base or open-source projects

I think it's pretty simple, I hope you will confirm 😅

It's about looking for a block that would validate expected format.

Right now

```go
package sandbox

// whatever the comment

func Foo() {} MATCH /should have a comment or be unexported/

// Bar is not OK because of the extra blank line

func Bar() {} // MATCH /should have a comment or be unexported/

// yellow is ...
func Qux() {} // MATCH /should be form Qux.../

```

Suggestion

```go
package sandbox

// whatever the comment

func Foo() {} MATCH /Foo should have a comment or be unexported/

// Bar is not OK because of the extra blank line

func Bar() {} // MATCH /extra blank line found blah blah/

// yellow is ...
func Qux() {} // MATCH /should be form Qux.../
```

Only Bar will be affected

**Comment by chavacava** [comment]
The proposal LGTM, though IMO it's low priority and shall be implemented in a way that the added complexity is justified (i.e. should be implemented in a very simple way)
